File: CWellness/src/main/java/com/cwsdg/cwsdg/service/UserService.java
package com.cwsdg.cwsdg.service;

import com.cwsdg.cwsdg.controller.model.UserDTO;
import com.cwsdg.cwsdg.mapper.UserMapper;
import com.cwsdg.cwsdg.repository.UserRepository;
import com.cwsdg.cwsdg.repository.entity.User;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.ArrayList;
import java.util.List;
import java.util.stream.Collectors;

@RequiredArgsConstructor
@Service
public class UserService {

    private final UserRepository userRepository;

    public void createUser(String name) {

        User user = new User();
        user.setName(name);

        userRepository.save(user);

    }

    public List<UserDTO> buscarUsuario() {
        List<User> userList = userRepository.findAll();

        return userList.stream().map(UserMapper.INSTANCE::entityToDto).collect(Collectors.toList());
    }


}

File: src/main/kotlin/com/example/testskotlin/service/UserService.kt
package com.example.testskotlin.service

import com.example.testskotlin.repository.UserRepository
import com.example.testskotlin.repository.model.User
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class UserService {

    @Autowired
    lateinit var userRepository: UserRepository

    fun getUsers(): User {
        return userRepository.findAll().first();
    }

}